pkg/cstorpool/v1alpha2/kubernetes.go

```diff
@@ -88,24 +153,71 @@ func KubeClient(opts ...kubeclientBuildOption) *kubeclient {
 
 // getClientOrCached returns either a new instance
 // of kubernetes client or its cached copy
-func (k *kubeclient) getClientOrCached() (*clientset.Clientset, error) {
+func (k *Kubeclient) getClientOrCached() (*clientset.Clientset, error) {
 	if k.clientset != nil {
 		return k.clientset, nil
 	}
 	c, err := k.getClientset()
 	if err != nil {
-		return nil, err
+		return nil,
+			errors.WithStack(errors.Wrapf(err, "failed to get clientset:"))
```

Comment:
We can use errors.Wrap instead of errors.Wrapf here.

Reply:
done.